### DISCUSSION TITLE
Textbox fill Grid column?

### DISCUSSION THREAD

**Discussion by renrutsirhc** [AvaloniaUI/Avalonia#5916] [Q&A] [answered]
Hi all,
I have an ItemsControl containing a grid. The question is, how do I make the textboxes in column 4 fill the available space? I've seen references to adding the HorizontalContentAlignment="Stretch" property to the ItemsControl in similar WPF scenarios but Avalonia does not seem to support this for ItemsControls. Any insight appreciated!
Cheers
Chris

![image](https://user-images.githubusercontent.com/12220071/118086419-0e79a500-b418-11eb-85ac-5e60f2c45653.png)

```
 <ItemsControl Items="{Binding Sections}" Grid.IsSharedSizeScope="True" Padding="2" Margin="2" HorizontalAlignment="Stretch">
              <ItemsControl.ItemTemplate>
                <DataTemplate>
                  <Grid Margin="0" HorizontalAlignment="Stretch">
                    <Grid.ColumnDefinitions>
                      <ColumnDefinition SharedSizeGroup="A" Width="Auto"/>
                      <ColumnDefinition Width="Auto"/>
                      <ColumnDefinition SharedSizeGroup="B" Width="135"/>
                      <ColumnDefinition Width="Auto"/>
                      <ColumnDefinition SharedSizeGroup="C" Width="Auto"/>
                    </Grid.ColumnDefinitions>
                    <TextBlock Grid.Column="0" VerticalAlignment="Center" Text="{Binding SectionName}"></TextBlock>
                    <NumericUpDown Grid.Column="2" Margin="2" Minimum="0" ButtonSpinnerLocation="Right" Increment="1" Value="{Binding NumberOfPlayersInSection}"></NumericUpDown>
                    <TextBox Grid.Column="4" Margin="2" Text="{Binding InstrumentDoublings}" HorizontalAlignment="Stretch"></TextBox>
                    <GridSplitter Grid.Column="1" HorizontalAlignment="Center" Width="1" VerticalAlignment="Stretch" Background="LightGray" Margin="10 0 10 0"></GridSplitter>
                    <GridSplitter Grid.Column="3" HorizontalAlignment="Center" Width="1" VerticalAlignment="Stretch" Background="LightGray" Margin="10 0 10 0"></GridSplitter>
                  </Grid>
                </DataTemplate>
              </ItemsControl.ItemTemplate>
            </ItemsControl>
```
EDIT
Hi all,
So this was working great with the solution provided by @workgroupengineering. I changed the ItemsControl to be a Listbox and then it wasn't working anymore. Is there any reason why the behaviour would be different between ItemsControl and ListBox for this?
Thanks so much!
Chris
EDIT 2
OK - figured this out - the ListBoxItem has the HorizontalContentAlignment Property which is set to Left by default - set this to stretch and all is well!

```
              <ItemsControl.Styles> 
                <Style Selector="ListBoxItem">
                  <Setter Property="HorizontalContentAlignment" Value="Stretch"/>
                </Style>
            </ItemsControl.Styles> 
```

**Comment by maxkatz6** [comment, marked as answer]
For the last column there should be Width="*" instead of Auto.

**Reply by renrutsirhc** [reply]
Thanks @maxkatz6 ! Tried that and getting exactly the same result. Any ideas?

**Reply by maxkatz6** [reply]
The easiest way to find out is to debug with devtools (press F12 in the running app). So you will be able to see which control is not stretching.

**Reply by renrutsirhc** [reply]
Oh that's neat! Had no idea you could do that! OK, so everything down to and including the Grid is showing the full width.

**Reply by maxkatz6** [reply]
Don't forget to check visual tree, not the logical tree. 
I can't help right now though, don't have access to the PC.

**Reply by renrutsirhc** [reply]
Yep - both showing the same thing! Thanks for the hints so far!

**Reply by workgroupengineering** [reply]
I think it's the same behavior as WPF, you need to remove SharedSizeGroup = "C" and set Width = "*".

**Reply by renrutsirhc** [reply]
@workgroupengineering Bingo! That's the winning combo. Thanks!
